code/modules/paperwork/pen.dm

```diff
@@ -133,6 +171,9 @@
 	..()
 	if(current_skin)
 		desc = "It's an expensive [current_skin] fountain pen. The nib is quite sharp."
+		var/datum/component/dart_insert/insert_comp = GetComponent(/datum/component/dart_insert)
+		insert_comp.casing_overlay_icon_state = overlay_reskin[current_skin]
+		insert_comp.projectile_overlay_icon_state = "[overlay_reskin[current_skin]]_proj"
```

Comment:
Can't ``reskin_obj`` send a signal instead of using GetComponent?

Reply:
I didn't see a signal being sent in the base implementation of `reskin_obj`, if that's what you mean.

Reply:
I'm just thinking it would be better if it did, so more components can take advantage of its parent item being reskinned, instead of doing a getcomponent here.